lib/adagrams.rb

```diff
@@ -0,0 +1,123 @@
+require 'csv'
+
+##################### wave 1
+def draw_letters
+  letter_pool = []
+
+  9.times {letter_pool.push("A", "I")}
+  2.times {letter_pool.push("B", "C", "F", "H", "M", "P", "V", "W", "Y" )}
+  4.times {letter_pool.push("D", "L", "S", "U")}
+  12.times {letter_pool << "E"}
+  3.times {letter_pool << "G"}
+  letter_pool << "J"
+  letter_pool << "K"
+  6.times {letter_pool.push("N", "R", "T")}
+  8.times {letter_pool << "O"}
+  letter_pool.push("Q", "X", "Z")
+
+  drawn_letters = letter_pool.sample(10)
+  return drawn_letters
+end
+
+####################### wave 2
+
+def uses_available_letters?(input, letters_in_hand)
+  input_letters = input.split("")
+  if input_letters.length > 10
+    puts "Oops, too many letters! You only have 10 :)"
```

Comment:
You don't need to do `puts` here.  Just return false or raise an error.   Probably raising an error makes sense here. 

```suggestion
```